apis/k8ssandra/v1alpha1/k8ssandracluster_types.go

```diff
@@ -30,6 +30,14 @@ import (
 
 // K8ssandraClusterSpec defines the desired state of K8ssandraCluster
 type K8ssandraClusterSpec struct {
+
+	// Whether to enable authentication in this cluster. The default is true; it is highly recommended to always leave
+	// authentication turned on. When enabled, authentication will be enforced not only on Cassandra nodes, but also on
+	// Reaper, Medusa and Stargate nodes, if any.
+	// +optional
+	// +kubebuilder:default=true
+	Auth bool `json:"auth"`
```

Comment:
TIL: you can't set `omitempty` here because that would cause the Go client to not include `auth:false` in the JSON payload, even if you explicitly set `Auth: false` in the struct. I guess this is because `false` is the zero value for `bool`.

Reply:
Actually it's even trickier. I think it's impossible to have a `bool` field with default value `true`. I'll switch to `*bool`.